flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -664,7 +740,7 @@ void testNumRestartsMetric() throws Exception {
         // wait for the second task submissions
         taskManagerGateway.waitForSubmissions(PARALLELISM);
 
-        assertThat(numRestartsMetric.getValue()).isEqualTo(1L);
+        assertThat(numRestartsMetric.getValue()).isEqualTo(0L);
```

Comment:
we seem to have removed the test for numRestartsMetric being incremented. I think we still need to test that - maybe in another test

Reply:
I need to second what @davidradl said, we need to test that the restart metrics still work.